src/sysext/sysext.c

```diff
@@ -784,6 +860,12 @@ static int merge(Hashmap *images) {
         if (r < 0)
                 return r;
 
+        if (!arg_no_reload && need_reload()) {
```

Comment:
please don't run this if exit code 123 is seen of the sd-merge child, see code above